server/channels/store/searchlayer/team_layer.go

```diff
@@ -33,15 +37,30 @@ func (s SearchTeamStore) UpdateMember(rctx request.CTX, teamMember *model.TeamMe
 func (s SearchTeamStore) RemoveMember(rctx request.CTX, teamId string, userId string) error {
 	err := s.TeamStore.RemoveMember(rctx, teamId, userId)
 	if err == nil {
-		s.rootStore.indexUserFromID(rctx, userId)
+		// Nothing to do if search engine is not active
+		if s.rootStore.searchEngine.ActiveEngine() != "database" && s.rootStore.searchEngine.ActiveEngine() != "none" {
+			s.rootStore.indexUserFromID(rctx, userId)
+			s.rootStore.indexChannelsForTeam(rctx, teamId)
+		}
 	}
 	return err
 }
 
 func (s SearchTeamStore) RemoveAllMembersByUser(rctx request.CTX, userId string) error {
-	err := s.TeamStore.RemoveAllMembersByUser(rctx, userId)
+	memberships, err := s.TeamStore.GetTeamsForUser(rctx, userId, "", true)
+	if err != nil {
+		return err
+	}
+	if s.rootStore.searchEngine.ActiveEngine() != "database" && s.rootStore.searchEngine.ActiveEngine() != "none" {
```

Comment:
Don't we have these strings ("database", "none") encoded in some constants or type?